ledger/src/shred/merkle.rs

```diff
@@ -307,6 +319,31 @@ impl ShredData {
         let node = get_merkle_node(shred, SIZE_OF_SIGNATURE..proof_offset).ok()?;
         get_merkle_root(index, node, proof).ok()
     }
+
+    pub(super) fn get_chained_merkle_root(
+        shred: &[u8],
+        proof_size: u8,
+        chained: bool,
+        resigned: bool,
+    ) -> Option<Hash> {
+        debug_assert_eq!(
+            shred::layout::get_shred_variant(shred).unwrap(),
+            ShredVariant::MerkleData {
+                proof_size,
+                chained,
+                resigned
+            }
+        );
+
+        if !chained {
+            return None;
+        }
+
+        let offset = Self::get_chained_merkle_root_offset(proof_size, resigned).ok()?;
+        shred
+            .get(offset..offset + SIZE_OF_MERKLE_ROOT)
+            .map(Hash::new)
+    }
```

Comment:
I am wondering if we need this function at all, or instead in `layout::get_chained_merkle_root` just call the `merkle::Shred{Data,Code}::get_chained_merke_root_offset` and do the slicing there.

This function is a bit ugly, and `layout::get_chained_merkle_root` only needs the offset, so maybe better not to add this function.